hare3/hare.go

```diff
@@ -323,15 +329,32 @@ func (h *Hare) onLayer(layer types.LayerID) {
 		return
 	}
 	h.patrol.SetHareInCharge(layer)
-	proto := newProtocol(h.config.Committee/2 + 1)
+
+	var (
+		signers []*signing.EdSigner
+		proto   = newProtocol(h.config.Committee/2 + 1)
+	)
 	h.mu.Lock()
+	// signer can't join mid session
+	signers = make([]*signing.EdSigner, 0, len(h.signers))
+	for _, signer := range h.signers {
+		signers = append(signers, signer)
+	}
```

Comment:
consider using `Values` from `exp/maps`:
```suggestion
	signers = maps.Values(h.signers)
```